test/runtests.jl

```diff
@@ -38,12 +38,47 @@ end
 
     @test a^(p - 1) == 1
     @test a^(-1) == inv(a)
+
+    @test Mod{p}(3//7) == 3//7
+    @test isequal(3//7,  Mod{p}(3//7))
+end
+
+@testset "Mod arithmetic with UInt" begin
+    p = UInt(23)
+    a = Mod{p,UInt}(2)
+    b = Mod{p,UInt}(25)
+    @test a == b
+    @test a == 2
+    @test a == 25
+
+    b = Mod{p,UInt}(20)
+    @test a + b == 22
+    @test a - b == 28
+    @test a + a == 2a
+    @test 0 - a == -a
+
+    @test a * b == Mod{p,UInt}(17)
+    @test (a / b) * b == a
+    @test (b // a) * (2 // 1) == b
+    @test a * (2 // 3) == (2a) * inv(Mod{p,UInt}(3))
+
+    @test is_invertible(a)
+    @test !is_invertible(Mod{10}(4))
+
+    @test a^(p - 1) == 1
+    @test a^(-1) == inv(a)
 end
 
 @testset "GaussMod arithmetic" begin
+    @test one(GaussMod{6}) == Mod{6}(1+0im)
+    @test zero(GaussMod{6}) == Mod{6}(0+0im)
+    @test GaussMod{6}(2 + 3im) == Mod{6}(2+3im)
+    @test GaussMod{6,Int}(2 + 3im) == Mod{6}(2+3im)
+    @test rand(GaussMod{6}) isa GaussMod
+    @test eltype(rand(GaussMod{6}, 4, 4)) <: GaussMod
     p = 23
-    a = GaussMod{p}(3 - im)
```

Comment:
Why do you ban `GaussMod` from the tests, it is stell defined and to be tested.

Reply:
I added this `GaussMod` back to make sure this PR is compatible with previous interface.

Reply:
It is required in the tests because it is defined in the src.

Reply:
Yeah, I meant I added GaussMod back to the tests.